### PR TITLE
Update to latest lsp-mode, lsp-ui

### PR DESCRIPTION
**Pull request by sebastiansturm**
This updates `lsp-mode` and `lsp-ui`, pulling in 2 performance improvements (https://github.com/emacs-lsp/lsp-mode/commit/da406084aef627eedaadeb94d96e1ddd53860673 and https://github.com/emacs-lsp/lsp-mode/commit/2fc0963a50488c47a5be3836ad042380b71d1cfb). There was one small breaking change introduced by https://github.com/emacs-lsp/lsp-mode/commit/4d59bded6bbba6d3bc73032305d1b3c071eb73c4 which I have tried to address; noticed no other breaking changes.
Opening this PR so other people can test it as well; I've only given it a few hours of superficial testing so far (using C++ and TypeScript language servers)

### CODE REVIEW

File: modules/tools/lsp/config.el [hunk 1]
```diff
@@ -27,8 +27,9 @@ This can be a single company backend or a list thereof. It can be anything
         lsp-intelephense-storage-path (concat doom-cache-dir "lsp-intelephense/"))
 
   :config
-  (when (and lsp-auto-configure lsp-auto-require-clients)
-    (require 'lsp-clients))
+  (when (and lsp-auto-configure)
+    (seq-do (lambda (package) (require package nil t))
+              lsp-client-packages))
```
Comment (hlissner) [inline modules/tools/lsp/config.el]: Please use `mapc`. Doom does not (explicitly) load the `seq` library in interactive sessions, so this is not guaranteed to work.

Comment (hlissner) [reply modules/tools/lsp/config.el]: Also, is there any possibility that a package in `lsp-client-packages` will fail to load? If not, `(mapc #'require lsp-client-packages)` is more concise.

File: modules/tools/lsp/config.el [hunk 2]
```diff
@@ -27,8 +27,9 @@ This can be a single company backend or a list thereof. It can be anything
         lsp-intelephense-storage-path (concat doom-cache-dir "lsp-intelephense/"))
 
   :config
-  (when (and lsp-auto-configure lsp-auto-require-clients)
-    (require 'lsp-clients))
+  (when (and lsp-auto-configure)
```
Comment (hlissner) [inline modules/tools/lsp/config.el]: Remove the `(and ...)` if there is only one argument.

Comment (sebastiansturm) [reply modules/tools/lsp/config.el]: ouch, that's a dumb one. Thanks for all your suggestions, will do that

File: modules/tools/lsp/config.el (hunk 1 above)
Comment (sebastiansturm) [reply modules/tools/lsp/config.el]: as to whether loading may fail: I think so; as far as I can see, `lsp-client-packages` contains several packages that are not part of `lsp-mode` but distributed separately (`lsp-python-ms`, `lsp-java`, `lsp-haskell`, maybe more)

File: modules/tools/lsp/config.el (hunk 2 above)
Comment (sebastiansturm) [reply modules/tools/lsp/config.el]: have made the changes (didn't squash commits though, should I?)

Comment (hlissner) [reply modules/tools/lsp/config.el]: @wildsebastian Yes, please do.

Comment (sebastiansturm) [reply modules/tools/lsp/config.el]: done

**Comment by hlissner** [comment]
Thanks for the help!

**Pull request merged**